src/test/java/org/gridsuite/modification/server/modifications/TwoWindingsTransformerCreationTest.java

```diff
@@ -0,0 +1,558 @@
+                                        .index(6)
+                                        .rho(1)
+                                        .r(0)
+                                        .x(0)
+                                        .g(0)
+                                        .b(0)
+                                        .build(),
+                                TapChangerStepCreationInfos.builder()
+                                        .index(7)
+                                        .rho(1)
+                                        .r(0)
+                                        .x(0)
+                                        .g(0)
+                                        .b(0)
+                                        .build(),
+                                TapChangerStepCreationInfos.builder()
+                                        .index(8)
+                                        .rho(1)
+                                        .r(0)
+                                        .x(0)
+                                        .g(0)
+                                        .b(0)
+                                        .build()
+                        ))
+                        .build())
+                .build();
+    }
+
+    @Override
+    protected MatcherTwoWindingsTransformerCreationInfos createMatcher(ModificationInfos modificationInfos) {
+        return MatcherTwoWindingsTransformerCreationInfos.createMatcherTwoWindingsTransformerCreationInfos((TwoWindingsTransformerCreationInfos) modificationInfos);
+    }
+
+    @Override
+    protected void assertNetworkAfterCreation() {
+        assertEquals(1, getNetwork().getVoltageLevel("v1").getTwoWindingsTransformerStream().filter(transformer -> transformer.getId().equals("new2wt")).count());
+        assertEquals(1, getNetwork().getVoltageLevel("v2").getTwoWindingsTransformerStream().filter(transformer -> transformer.getId().equals("new2wt")).count());
+        assertEquals("v1", getNetwork().getTwoWindingsTransformer("new2wt").getTerminal1().getVoltageLevel().getId());
+        assertEquals("v2", getNetwork().getTwoWindingsTransformer("new2wt").getTerminal2().getVoltageLevel().getId());
+        assertEquals(2., getNetwork().getTwoWindingsTransformer("new2wt").getX(), 0.1);
+        assertEquals(5., getNetwork().getTwoWindingsTransformer("new2wt").getRatedU1(), 0.1);
+        assertEquals(1, getNetwork().getTwoWindingsTransformer("new2wt").getRatedS(), 0.1);
+        assertEquals(4, getNetwork().getTwoWindingsTransformer("new2wt").getRatioTapChanger().getStepCount());
+        assertEquals(3, getNetwork().getTwoWindingsTransformer("new2wt").getPhaseTapChanger().getStepCount());
+        assertEquals(PhaseTapChanger.RegulationMode.CURRENT_LIMITER, getNetwork().getTwoWindingsTransformer("new2wt").getPhaseTapChanger().getRegulationMode());
+    }
+
+    @SneakyThrows
+    @Override
+    protected void assertNetworkAfterDeletion() {
```

Comment:
you can also put back :
`assertNull(getNetwork().getTwoWindingsTransformer("new2wt"));`